main/ts_serial.c

```diff
@@ -199,4 +197,79 @@ void ts_serial_response_clear()
     xSemaphoreGive(resp_buf_lock);
 }
 
+// CAN_Address is not needed here, but we need the same signature
+// as CAN send
+char *ts_serial_send(char *req, uint8_t CAN_Address)
+{
+    if (req == NULL) {
+        ESP_LOGE(TAG, "Got invalid parameter");
+        return NULL;
+    }
+
+    if (ts_serial_request(req, 200) != ESP_OK) {
+        ESP_LOGE(TAG, "Request failed: %s", req);
+        return NULL;
+    }
+
+    char *buf = ts_serial_response(200);
+    if (buf == NULL) {
+        ESP_LOGE(TAG, "Response failed");
+        ts_serial_response_clear();
+        return NULL;
+    }
+
+    char *resp = (char *) heap_caps_malloc(strlen(buf)+1, MALLOC_CAP_8BIT);
+    strcpy(resp, buf);
+    ts_serial_response_clear();
+    return resp;
+}
+
+int ts_serial_scan_device_info(TSDevice *device)
+{
+    char req[7]= "?info\n\0";
+    // First request mostly fails, so we request it twice
+    if (ts_serial_request(req, 500) == ESP_FAIL) {
+        ESP_LOGE(TAG, "Could not scan for devices on serial adapter");
+    }
+
+    char *resp = ts_serial_response(500);
+    int status = ts_resp_status(resp != NULL ? resp : "");
+    if (status != TS_STATUS_CONTENT) {
+        ESP_LOGE(TAG, "Could not retrieve device information: Code %d", status);
+        ts_serial_response_clear();
+    }
+
+    ts_serial_response_clear();
+
+    if (ts_serial_request(req, 500) == ESP_FAIL) {
+        ESP_LOGE(TAG, "Could not scan for devices on serial adapter");
+        return -1;
+
+    }
+
+    resp = ts_serial_response(500);
+    status = ts_resp_status(resp != NULL ? resp : "");
+    if (status != TS_STATUS_CONTENT) {
+        ESP_LOGE(TAG, "Could not retrieve device information: Code %d", status);
+        ts_serial_response_clear();
+        return -1;
+    }
+
+    // shift pointer to data
+    resp = ts_resp_data(resp);
+
+    cJSON *json_data = cJSON_Parse(resp);
+    ts_serial_response_clear();
+    device->ts_name = (char *) heap_caps_malloc(strlen(cJSON_GetStringValue(cJSON_GetObjectItem(json_data, "DeviceType")))+1, MALLOC_CAP_8BIT);
```

Comment:
Line too long and hard to read. Suggesting

```suggestion
    size_t ts_name_len = strlen(cJSON_GetStringValue(cJSON_GetObjectItem(json_data, "DeviceType")));
    device->ts_name = (char *)heap_caps_malloc(ts_name_len + 1, MALLOC_CAP_8BIT);
```

Same also below.